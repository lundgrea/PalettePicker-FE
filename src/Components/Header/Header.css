.Header {
  height: 100px;
  width: 100%;
  border-bottom: 2px solid black;
  font-family: 'Caveat', cursive;
  display: flex;
  justify-content: flex-end;
}

#header-text {
  margin-right: 20px;
  font-size: 50px;
}