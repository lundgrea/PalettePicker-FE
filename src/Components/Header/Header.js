import React from 'react'
import './Header.css'

export const Header = (props) => {
  return (
    <header className="Header">
      <button 
      id="header-text-button"
      onClick={e => props.clearFolderID(e)}>Happy Little Palette Picker</button>
    </header>
  )
}

export default Header
