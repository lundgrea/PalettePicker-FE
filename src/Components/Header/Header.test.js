import React from "react";
import { shallow } from "enzyme";
import { Header } from "./Header";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Header clearFolderID={jest.fn()}/>)
});


describe("Header", () => {
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call clearFolderID on button click', () => {
    const clearFolderIDMock = jest.fn()
    wrapper = shallow(<Header clearFolderID={clearFolderIDMock}/>)
    const mockEvent = { preventDefault: jest.fn()}
    wrapper.find('button').simulate('click', mockEvent)
    expect(clearFolderIDMock).toHaveBeenCalledWith(mockEvent)
  })
});

