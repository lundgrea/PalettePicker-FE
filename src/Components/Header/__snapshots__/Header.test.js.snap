// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  className="Header"
>
  <button
    id="header-text-button"
    onClick={[Function]}
  >
    Happy Little Palette Picker
  </button>
</header>
`;
